compiler_opt/es/blackbox_optimizers.py

```diff
@@ -263,7 +277,141 @@ def run_step(self, perturbations: npt.NDArray[np.float32],
     # gradients
     return self.gradient_ascent_optimizer.run_step(current_input, gradient)
 
-  def get_hyperparameters(self):
+  def get_hyperparameters(self) -> List[float]:
+    if self.hyperparameters_update_method == UpdateMethod.STATE_NORMALIZATION:
+      return self.mean_state_vector + self.std_state_vector
+    else:
+      return []
+
+  def get_state(self) -> List[float]:
+    gradient_ascent_state = self.gradient_ascent_optimizer.get_state()
+    if self.hyperparameters_update_method == UpdateMethod.STATE_NORMALIZATION:
+      current_state = [self.nb_steps]
+      current_state += self.sum_state_vector
+      current_state += self.squares_state_vector
+      current_state += self.mean_state_vector + self.std_state_vector
+      return current_state + gradient_ascent_state
+    else:
+      return gradient_ascent_state
+
+  def update_state(self, evaluation_stats: SequenceOfFloats) -> None:
+    if self.hyperparameters_update_method == UpdateMethod.STATE_NORMALIZATION:
```

Comment:
and if not, then what?
i.e. is it a bug to call `update_state` if self.hyperparameters_update_method != UpdateMethod.STATE_NORMALIZATION, or the API is meant to be callable in all cases, for uniformity, and only does something in this case?

if the latter, maybe easier to read if:
```
if self.<blah blah> != UpdateMethod.STATE_NORMALIZATION:
  return

<current code>
```

less indenting :)

Reply:
The latter is true so I will update the code to this style.